packages/transaction-controller/src/helpers/PendingTransactionTracker.ts

```diff
@@ -46,6 +57,13 @@ export class PendingTransactionTracker {
     this.#getEthQuery = getEthQuery;
     this.#getTransactions = getTransactions;
     this.#nonceTracker = nonceTracker;
+
+    this.#beforePublish = hooks?.beforePublish
+      ? hooks.beforePublish
+      : () => true;
+    this.#beforeCheckPendingTransaction = hooks?.beforeCheckPendingTransaction
```

Comment:
Are we missing the usages of these two?

Reply:
Rebased after your refactor in the `PendingTransactionTracker` and included those hooks in the proper places and unit tests.